src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -661,20 +661,93 @@ static int _xccdf_item_recursive_gather_selected_rules(struct xccdf_policy *poli
+		const char *profile_id = xccdf_profile_get_id(profile);
+		// Title
+		struct oscap_text_iterator *title_iterator = xccdf_profile_get_title(profile);
+		const char *profile_title = oscap_textlist_get_preferred_plaintext(title_iterator, NULL);
+		oscap_text_iterator_free(title_iterator);
+		// Description
+		struct oscap_text_iterator *description_iterator = xccdf_profile_get_description(profile);
+		const char *profile_description = description_iterator != NULL ?
+				oscap_textlist_get_preferred_plaintext(description_iterator, NULL) : "Not available";
+		oscap_text_iterator_free(description_iterator);
+
+		struct xccdf_benchmark *benchmark = xccdf_policy_get_benchmark(policy);
+		const char *benchmark_version_info = xccdf_benchmark_get_version(benchmark);
+		const char *benchmark_id = xccdf_benchmark_get_id(benchmark);
+		const struct xccdf_version_info *xccdf_version = xccdf_benchmark_get_schema_version(benchmark);
+		const char *xccdf_version_name = xccdf_version_info_get_version(xccdf_version);
+
+		fix_header = oscap_sprintf(
+			"###############################################################################\n#\n"
+			"# Bash remediation role for profile %s\n"
+			"# Profile Title:  %s\n"
+			"# Profile Description:  %s\n"
+			"#\n"
+			"# Benchmark ID:  %s\n"
+			"# Benchmark Version:  %s\n#\n"
+			"# XCCDF Version:  %s\n#\n"
+			"# This file was generated by OpenSCAP %s using:\n"
+			"# 	$ oscap xccdf generate fix --profile %s%s%s sds.xml \n#\n"
+			"# This script is generated from an OpenSCAP profile without preliminary evaluation.\n"
+			"# It attempts to fix every selected rule, even if the system is already compliant.\n#\n"
+			"###############################################################################\n\n\n",
+				profile_id, profile_title, profile_description, benchmark_id, benchmark_version_info,
+				xccdf_version_name, oscap_version, profile_id, template, format);
 
 	} else {
-		// no header required
-		return 0;
+		// Results-based remediation fix
+		const char *start_time = xccdf_result_get_start_time(result);
+		const char *end_time = xccdf_result_get_end_time(result);
+		const char *result_id = xccdf_result_get_id(result);
+		const struct xccdf_version_info *xccdf_version = xccdf_result_get_schema_version(result);
+		const char *xccdf_version_name = xccdf_version_info_get_version(xccdf_version);
+
+		fix_header = oscap_sprintf(
+			"###############################################################################\n#\n"
+			"# Bash remediation role for the results of evaluation of profile %s \n"
+			"# XCCDF Version:  %s\n#\n"
+			"# Evaluation Start Time:  %s\n"
+			"# Evaluation End Time:  %s\n#\n"
+			"# This file was generated by OpenSCAP %s using one of the following:\n"
+			"# 	$ oscap xccdf generate fix --result-id %s%s%s xccdf-results.xml \n"
+			"# 	$ oscap xccdf generate fix --result-id %s%s%s arf.xml \n#\n"
+			"# This script is generated from the results of a profile evaluation.\n"
```

Comment:
So I've been looking into how to do this, and i either would have to pass the filename into the xccdf_policy_generate_fix as a parameter (which causes all sorts of issues with the python wrapper) or add some kind of getter/setter into one of the structures (policy, benchmark, etc)